compiler/typeallowed.nim

```diff
@@ -96,6 +96,8 @@ proc typeAllowedAux(marker: var IntSet, typ: PType, kind: TSymKind,
       result = nil
   of tyUntyped, tyTyped:
     if kind notin {skParam, skResult} or taNoUntyped in flags: result = t
+  of tyIterable:
+    if kind notin {skParam}: result = t
```

Comment:
Isn't it even worse than that? `tyIterable` is only valid for template/macro parameter types, right? --> We need a spec addition to clarify these things.

Reply:
I've now fixed this, `proc fn(a: iterable[int]) = discard` now gets rejected early (=here) instead of later